seqr/views/utils/terra_api_utils.py

```diff
@@ -0,0 +1,175 @@
+"""
+This module provides python bindings for the AnVIL Terra API.
+"""
+
+import json
+import logging
+
+from urllib.parse import urljoin
+
+from google.auth.transport.requests import AuthorizedSession
+from google.oauth2 import service_account
+
+from settings import SEQR_VERSION, TERRA_API_CONFIG
+
+SEQR_USER_AGENT = "seqr/" + SEQR_VERSION
+
+scopes = ['https://www.googleapis.com/auth/userinfo.profile',
+          'https://www.googleapis.com/auth/userinfo.email',
+          'https://www.googleapis.com/auth/cloud-billing',
+          'openid']
+
+logger = logging.getLogger(__name__)
+
+
+class TerraAPIException(Exception):
+    """For exceptions happen in Terra API calls
+    """
+    pass
+
+
+class AnvilSession:
+    def __init__(self, credentials=None, service_account_secret="service_account.json", scopes=None):
+        if credentials is None:
+            credentials = service_account.Credentials.from_service_account_file(service_account_secret, scopes = scopes)
+        self._session = AuthorizedSession(credentials = credentials)
+
+    #################################################
+    # Utilities
+    #################################################
+    def _seqr_agent_header(sefl, headers=None):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=6211929)